Tests/SentryTests/Performance/SentryTracerTests.swift

```diff
@@ -136,11 +150,36 @@ class SentryTracerTests: XCTestCase {
             XCTAssertEqual(tracerTimestamp.timeIntervalSince1970, span["timestamp"] as? TimeInterval)
         }
     }
-    
+
+    func testDeadlineTimer_FinishesTransactionAndChildren() {
+        let sut = fixture.getSut()
```

Comment:
I would feel safer if we added a test that a manually created transaction doesn't time out. We can do that by validating that the SentryTracer doesn't interact with the `timerWrapper` for manually created transactions.

Reply:
Ok, I added `testDeadlineTimer_OnlyForAutoTransactions`